ogr/abstract.py

```diff
@@ -217,7 +217,7 @@ def body(self, new_body: str) -> None:
         self._body = new_body
 
     @property
-    def id(self) -> int:
+    def id(self) -> Optional[int]:
```

Comment:
Does it make sense though?

I think I know what you meant in the description.

Reply:
Yes, this is a bit of a dilemma. It doesn't make much sense semantically, however just based on our interface, this could happen. If I wanted to enforce an `int` here, the order of arguments in the constructor would have to be changed (along with some other changes of course)